pramen/core/src/main/scala/za/co/absa/pramen/core/sql/SqlGeneratorBase.scala

```diff
@@ -49,66 +68,95 @@ abstract class SqlGeneratorBase(sqlConfig: SqlConfig) extends SqlGenerator {
     if (columns.isEmpty) {
       "*"
     } else {
-      columns.map(col => escapeIdentifier(col)) .mkString(", ")
+      columns.map(col => escape(col)) .mkString(", ")
+    }
+  }
+
+  final def splitComplexIdentifier(identifier: String): Seq[String] = {
+    val trimmedIdentifier = identifier.trim
+
+    if (trimmedIdentifier.isEmpty) {
+      throw new IllegalArgumentException(f"Found an empty table name or column name ('$identifier').")
     }
+
+    val (escapeBegin, escapeEnd) = beginEndEscapeChars
+    val sameEscapeChar = escapeBegin == escapeEnd
+
+    val output = new ListBuffer[String]
+    val curColumn = new StringBuffer()
+    val len = trimmedIdentifier.length
+    var nestingLevel = 0
+    var i = 0
+
+    while (i < len) {
```

Comment:
I'll try, thanks. But from the looks of it, I'm not sure if `current` ever becomes the first character and the last character of the string, but it is important that it should.

Reply:
Sorry, I've missed the padding at the beginning :)
Yes, it definitely works, and I didn't know about sliding() - it is an elegant solution.

But I did a micro-benchmark for strings having length of 60 characters. The results are:
- The current solution: 663 nano seconds
- The sliding window solution: 6971 nano seconds (more than 10 times slower)
Although the code is not on the critical path, the method is general enough that it can be reused elsewhere, and I'd stick with the more performant one. 

The code of the benchmark:
<details>
<summary>Click to expand</summary>

```scala

val testStr = "abc"*20

def test1(str: String): Unit = {
  val paddedString = " " + str + " "

  val buf = new StringBuffer()
  paddedString.sliding(3).zipWithIndex.foreach { case (window, i) =>
    val prev = window(0)
    val current = window(1)
    val next = window(2)

    buf.append(current)
  }
}

def test2(str: String): Unit = {
  val buf = new StringBuffer()

  var i = 0
  while (i < str.length) {
    val prev = if (i > 0) str(i-1) else ' '
    val current = str(i)
    val next = if (i < str.length - 1) str(i + 1) else ' '

    buf.append(current)
    i += 1
  }
}

val N = 1000000

val start1 = System.nanoTime()
var i = 0
while (i < N) {
  test1(testStr)
  i += 1
}
val end1 = System.nanoTime()
println(s"test1: ${(end1 - start1)/N} nano s")

val start2 = System.nanoTime()
i = 0
while (i < N) {
  test2(testStr)
  i += 1
}
val end2 = System.nanoTime()
println(s"test2: ${(end2 - start2)/N} nano s")
```

</details>

Reply:
Expert answer 😄